tests/RA_StringUtils_Tests.cpp

```diff
@@ -46,12 +44,22 @@ TEST_CLASS(RA_StringUtils_Tests)
 
     TEST_METHOD(TestTrimLineEnding)
     {
-        Assert::AreEqual(std::string("test"), TrimLineEnding(std::string("test")));
-        Assert::AreEqual(std::string("test"), TrimLineEnding(std::string("test\r")));
-        Assert::AreEqual(std::string("test"), TrimLineEnding(std::string("test\n")));
-        Assert::AreEqual(std::string("test"), TrimLineEnding(std::string("test\r\n")));
-        Assert::AreEqual(std::string("test\n"), TrimLineEnding(std::string("test\n\n")));
-        Assert::AreEqual(std::string("test\r\n"), TrimLineEnding(std::string("test\r\n\r\n")));
+        std::array<std::string, 6> test_strings
+        {
+            "test",
+            "test\r",
+            "test\n",
+            "test\r\n",
+            "test\n\n",
+            "test\r\n\r\n"
+        };
+
+        Assert::AreEqual(test_strings.at(0), TrimLineEnding(test_strings.at(0)));
+        Assert::AreEqual(test_strings.at(0), TrimLineEnding(test_strings.at(1)));
+        Assert::AreEqual(test_strings.at(0), TrimLineEnding(test_strings.at(2)));
+        Assert::AreEqual(test_strings.at(0), TrimLineEnding(test_strings.at(3)));
+        Assert::AreEqual(test_strings.at(2), TrimLineEnding(test_strings.at(4)));
+        Assert::AreEqual(test_strings.at(3), TrimLineEnding(test_strings.at(5)));
```

Comment:
This is where the` /W4` warnings were triggered. The param is supposed be `std::string&` but previously it was `std::string&&`, temporary variables do not go out, they get destroyed. Scopes are very important in object-oriented languages. You could use an `rvalue reference` instead of a `const lvalue reference`, but if it's not `const `it's undefined unless an extension adjusts for it.